src/link/assign.c

```diff
@@ -214,6 +216,56 @@ FindLargestSection(enum eSectionType type, bool bankFixed)
 	return r;
 }
 
+int
+IsSectionSameTypeBankAndFloating(const char *name, enum eSectionType type, int bank)
+{
+	struct sSection *pSection;
+
+	pSection = pSections;
+	while (pSection) {
+		if (pSection->oAssigned == 0) {
+			if (strcmp(pSection->pzName, name) == 0) {
+				if (pSection->nOrg != -1 || pSection->nAlign != 1)
+					return 0; /* not floating */
+				if (pSection->Type != type)
+					return 0; /* different type */
+				/* Bank number must be unassigned or equal */
```

Comment:
This is *tremendously* picky, but it'd be nice if all of the comments here were consistent, e.g. all above the `if`, or all after the `return`. Feel free to ignore this if you're not fussed 😝